src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java

```diff
@@ -48,6 +48,7 @@ protected void checkFile(@NotNull ErlangFile file, @NotNull ProblemsHolder probl
       List<ErlangCallbackSpec> undefinedCallbacks = ContainerUtil.newArrayList();
       Map<String, ErlangCallbackSpec> callbackMap = behaviourModule.getCallbackMap();
       for (ErlangCallbackSpec spec : callbackMap.values()) {
+        if (spec.isOptional()) continue;
```

Comment:
Optional callback attribute should be taken into account for release ```18.0``` and above.